src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2372,6 +2406,63 @@ LogicalResult ONNXConcatOp::inferShapes() {
         j == axisIndex ? cummulativeAxisSize : commonShape[j]);
   getResult().setType(
       RankedTensorType::get(outputDims, commonType.getElementType()));
+
+  // Checking to see if shape folding is possible
+  bool canShapeFold = false;
+  int64_t size = 0;
+  for (int i = 0; i < inputNum; ++i) {
+    if (auto constantOp = getONNXConstantOp(getOperand(i))) {
+      if (DenseElementsAttr shapeAttributes =
+              constantOp.valueAttr().dyn_cast_or_null<DenseElementsAttr>()) {
+        size += shapeAttributes.getValues<IntegerAttr>().end() -
+                shapeAttributes.getValues<IntegerAttr>().begin();
+      } else {
+        return emitError("Failed passing shape folding information");
```

Comment:
We do best-effort shape folding right now, so let's not error out here, preventing shape propagation.

Reply:
I am not sure about this one, this occurs when the constantop doesn't have a value set. I don't think you can do anything here.